### PR TITLE
Feature/query microservices

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

File: api/src/main/java/datawave/security/authorization/ProxiedUserDetails.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import java.util.Collection;
 import java.util.List;
+import java.util.function.Function;
```
Comment (jwomeara) [inline api/src/main/java/datawave/security/authorization/ProxiedUserDetails.java]: This probably isn't needed.  I can remove it.

**Review by jwomeara — COMMENTED** [review]
Notable changes in this PR:
 - Added direct dependency on accumulo
 - Moved the AuthorizationsPredicate logic here so it would be available to both the webservice and query microservices
 - Added listEffectiveAuthorizations endpoint (which technically was introduced with spring boot 2.x) and added support for federation of this request to downstream systems
 - Replicating some of the functionality found in UserOperationsBean for the query microservices
 - AuthorizationsListSupplier was moved to the datawave microservice starter since multiple services now need it

File: service/src/main/java/datawave/microservice/authorization/AuthorizationOperationsV1.java [hunk 2]
```diff
@@ -98,10 +122,32 @@ public AuthorizationsListBase<?> listEffectiveAuthorizations(@AuthenticationPrin
...
+        log.info("Flushing credentials for " + currentUser + " from the cache.");
+        
+        // if we have any remote services configured, then flush those credentials as well
+        if (federate) {
+            for (UserOperations federatedUserOperations : registeredFederatedUserOperations) {
+                try {
+                    federatedUserOperations.flushCachedCredentials(currentUser);
+                } catch (Exception e) {
+                    log.error("Failed to flush user from remote user service", e);
+                    response.addMessage("Unable to user from remote user service");
```
Comment (foster33) [inline service/src/main/java/datawave/microservice/authorization/AuthorizationOperationsV1.java]: Should this be unable to flush user?
```suggestion
                    response.addMessage("Unable to flush user from remote user service");
```

**Review by apmoriarty — APPROVED** [review]

**Review by alerman — APPROVED** [review]

File: service/src/main/java/datawave/microservice/authorization/AuthorizationOperationsV1.java (hunk 2 above)
Comment (jwomeara) [reply service/src/main/java/datawave/microservice/authorization/AuthorizationOperationsV1.java]: yes.  good catch.

**Review by matthpeterson — APPROVED** [review]

**Pull request merged**